City-of-Helsinki/drupal-helfi-platform-config
# UHF-7449 announcement sync

# Announcement sync [UHF-7449](https://helsinkisolutionoffice.atlassian.net/browse/UHF-7449)

## What was done
Added new block which utilises external entities
Added new external entity type for announcements

## How to install
* [Review this etusivu PR first](https://github.com/City-of-Helsinki/drupal-helfi-etusivu/pull/290)
* Make sure you have this line in your `local.settings.php`: `$config['helfi_announcement.settings']['source_environment'] = 'local';`
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update the Helfi Platform config
    * `composer require drupal/helfi_platform_config:dev-UHF-7449_announcement_sync`
* Run `make drush-updb drush-cr`

## How to test
If you tested the `etusivu PR` first, you should already have one or two global announcements created
- Check that you have `external announcements` block set up above `announcements block`
- Go to any page
  - You should see the global announcements created in etusivu above the site's own announcements.
    - Test with languages and  as well, and different announcement types( alert, notice...).
- Go back to `etusivu` and unpublish one or more global announcements
- Go back to original environment and wait for 10 minutes before running `drush cron`
  - unpublished external announcements should disappear 


## Other PRs

[UHF-7449]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-7449?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
